### ISSUE TITLE
CWBI Test /location/group endpoint returning location group without an id

### ISSUE THREAD

**Issue by adamkorynta** [USACE/cwms-data-api#582]
```
curl -X 'GET' \
  'https://cwms-data-test.cwbi.us/cwms-data/location/group?location-category-like=RDL_Findley' \
  -H 'accept: application/json'
```

Response
```
[
  {
    "location-category": {
      "office-id": "MVS",
      "id": "RDL_Findley"
    },
    "assigned-locations": []
  }
]
```

The group-id field should be required and the client does not expect it to be missing.

**Comment by rma-rripken** [comment]
Weird.   The Category is in an office but the group has a null group-id and a null group-office-id.  How do those get created and what are they used for?

**Comment by adamkorynta** [comment]
I am also seeing an issue where I filter the category, but not the office and include assigned locations
```
curl -X 'GET' \
  'https://cwms-data-test.cwbi.us/cwms-data/location/group?include-assigned=true&location-category-like=Agency%20Aliases' \
  -H 'accept: application/json'
```

This returns an error code 500, but no body and just these headers:
```
 access-control-allow-credentials: true 
 access-control-allow-headers: x-requested-with,Content-Type,origin,authorization,accept,client-security-token 
 access-control-allow-methods: POST,GET,OPTIONS,DELETE,PUT 
 access-control-max-age: 600 
 connection: close 
 content-length: 0 
 content-security-policy: frame-src 'self' *.usace.army.mil *.sec.usace.army.mil *.ops.usace.army.mil *.cwbi.us *.rsgis.dev; frame-ancestors 'self' *.usace.army.mil *.sec.usace.army.mil *.ops.usace.army.mil *.cwbi.us *.rsgis.dev 
 content-type: application/json 
 date: Fri,15 Mar 2024 18:12:26 GMT 
 server: USA 
 strict-transport-security: max-age=31536000; includeSubDomains 
 vary: Origin 
 x-content-type-options: nosniff 
 x-frame-options: SAMEORIGIN,SAMEORIGIN 
 x-xss-protection: 1; mode=block 
```

I think it might be related?

**Comment by rma-rripken** [comment]
It might be related.  Hard to say.   I think the already merged pr resolved this issue.  If filter the category, but not the office and include assigned locations continues to 500, lets open a new issue.   I just tried your query on my local system and got results (didn't get a 500) but I also don't have ton of data in my db so its possible I just don't have a group that triggers.

**Comment by rma-rripken** [comment]
@MikeNeilson   On 4/18 Kevin was still seeing issues with this against CWBI test for the NWDM office.   Need to see if there is something weird in the Location Categories  - like maybe a category with a null ID?

**Comment by rma-rripken** [comment]
@MikeNeilson There is still some sort of weird data in HEC T7 NWDM database that isn't fixed by my previous pr.  It is triggering a NPE in Kevin testing leading to https://bitbucket.hecdev.net/projects/SC/repos/hec-monolith/pull-requests/403.  Would you be able to look at that NWDM database and see if there is anything obviously wrong with the Location category/group data in there?   Things like Location Category with a null ID or a Location Group with a null id?  Or maybe an empty id that wasn't null?

**Comment by MikeNeilson** [comment]
Yeah, Probably not today though, but I'll try. That said I can't even get simple data from the HEC T7 and I think it has to do with the 24.x.y Schema upgrade.
